lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -25,19 +25,38 @@ using namespace mlir;
 // SVModuleOp
 //===----------------------------------------------------------------------===//
 
+void SVModuleOp::build(mlir::OpBuilder &builder, mlir::OperationState &state,
+                       llvm::StringRef name, hw::ModuleType type) {
+  state.addAttribute(SymbolTable::getSymbolAttrName(),
+                     builder.getStringAttr(name));
+  state.addAttribute(getModuleTypeAttrName(state.name), TypeAttr::get(type));
+  state.addRegion();
+}
+
 void SVModuleOp::print(OpAsmPrinter &p) {
   p << " ";
+
+  // Print the visibility of the module.
+  StringRef visibilityAttrName = SymbolTable::getVisibilityAttrName();
+  if (auto visibility = (*this)->getAttrOfType<StringAttr>(visibilityAttrName))
+    p << visibility.getValue() << ' ';
+
   p.printSymbolName(SymbolTable::getSymbolName(*this).getValue());
   hw::module_like_impl::printModuleSignatureNew(p, getBodyRegion(),
                                                 getModuleType(), {}, {});
-  p.printOptionalAttrDictWithKeyword(getOperation()->getAttrs(),
-                                     getAttributeNames());
   p << " ";
   p.printRegion(getBodyRegion(), /*printEntryBlockArgs=*/false,
                 /*printBlockTerminators=*/true);
+
+  SmallVector<StringRef, 4> omittedAttrs = {"sym_name", "module_type",
+                                            visibilityAttrName};
+  p.printOptionalAttrDictWithKeyword((*this)->getAttrs(), omittedAttrs);
```

Comment:
I think this change is not required anymore? Also this changes the position of attributes and is that intentional?

Reply:
You are right. It is no need to preserve it.

Reply:
I will delete it soon. Thanks so much.